pytradfri/api/aiocoap_api.py

```diff
@@ -15,56 +17,73 @@
     TimeoutError,
 )
 from aiocoap.numbers.codes import Code
+from aiocoap.protocol import BlockwiseRequest
 
 from ..command import Command
 from ..error import ClientError, RequestTimeout, ServerError
 from ..gateway import Gateway
 
 _LOGGER = logging.getLogger(__name__)
-_SENTINEL = object()
+
+
+class UndefinedType(Enum):
+    """Singleton type for use with not set sentinel values."""
+
+    _singleton = 0
+
+
+_SENTINEL = UndefinedType._singleton  # pylint: disable=protected-access
```

Comment:
I have never seen this type of code before. What does it do?

Reply:
It's to type the singleton that is used to mark empty value.

https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions

Reply:
Thanks!